erpnext/stock/utils.py

```diff
@@ -660,20 +660,24 @@ def get_combine_datetime(posting_date, posting_time):
 	return datetime.datetime.combine(posting_date, posting_time).replace(microsecond=0)
 
 
+@frappe.request_cache
 def get_default_stock_uom() -> str | None:
-	default_uom = frappe.db.get_single_value("Stock Settings", "stock_uom")
-	if default_uom:
+	if default_uom := frappe.get_cached_value("Stock Settings", None, "stock_uom"):
 		return default_uom
 
-	if frappe.db.exists("UOM", "Nos"):
-		return "Nos"
+	acceptable_default_uoms = dict.fromkeys(
```

Comment:
@sagarvora why not create a tuple in the first place?

Reply:
dict.fromkeys creates an ordered set => removes duplicates